cmd/root.go

```diff
@@ -46,12 +46,21 @@ func New() *cobra.Command {
 connecting to Cloud SQL instances. It listens on a local port and forwards connections
 to your instance's IP address, providing a secure connection without having to manage
 any client SSL certificates.`,
-		RunE: runSignalWrapper,
+		// By default Cobra will print usage if RunE returns an error. Since we
+		// don't want to see usage on a normal SIGTERM exit, we'll manually
+		// print usage when no arguments are provided and otherwise disable this
+		// "feature."
+		SilenceUsage: true,
+		RunE:         runSignalWrapper,
 	}
 }
 
 // runSignalWrapper watches for SIGTERM and SIGINT and interupts execution if necessary.
 func runSignalWrapper(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		cmd.Help()
+		os.Exit(1)
+	}
```

Comment:
So I agree we don't always want the usage, but I think that only printing it when no args are set isn't maybe the best idea either. Passing an invalid arg will no longer trigger the help message here. 

Take a look at [this comment ](https://github.com/spf13/cobra/issues/340#issuecomment-374617413) - they recommended setting "SilenceUsuage" from within RunE after the args are parsed to avoid having it print for errors once a certain point is reached.

Reply:
Nice find. I'll update with that approach.